tests/framework/utils.py

```diff
@@ -515,3 +516,26 @@ def compare_versions(first, second):
         return first[1] - second[1]
 
     return first[0] - second[0]
+
+
+class ResultsFileDumper:
+    """Class responsible with outputting test results to files."""
+
+    def __init__(self, test_name: str):
+        """Initialize the instance."""
+        self._root_path = defs.TEST_RESULTS_DIR
+        self._path = self._root_path / test_name
+
+        # Create the root directory, if it doesn't exist.
+        os.makedirs(self._root_path, exist_ok=True)
+
+    def writeln(self, data: str, append=True):
```

Comment:
Do we ever need to append to a file? 

Seems like this is only called with json files as parameters and serializing jsons creates an invalid file.

Reply:
I don't quite understand what you mean by `serializing jsons creates an invalid file`. Writing a line of JSON to a file does not invalidate that file in any way. It's not binary serialized.

This method can be called with any string payload, depending on the test and the baselines parser. The fact that it uses JSON here is specific to this test.

Moreover, I added the default behaviour of appending to the file for the case where we want to run `devtool test ..` in a loop, gathering data from multiple iterations. Since we are appending to the file, one will not lose the data gathered from previous iterations.